### PR TITLE
Bindings for Block

### LINKED ISSUE #8: Bindings for Block
Implement Java bindings for `Block` and all `Block` related methods on `File`.

Write tests.

### PR DESCRIPTION
**Pull request by sujithvm**
This pull request contains commits to fix #8 . Thank you

### CODE REVIEW

**Comment by sujithvm** [comment]
@stoewer Can you review this pull request. Thank you

File: src/main/java/org/gnode/nix/Block.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        }
+        return null;
+    }
+
+    private native void definition(@Const @ByVal None t);
+
+    /**
+     * Deletes for the definition of the block.
+     */
+    public void removeDefinition() {
```
Comment (stoewer) [inline src/main/java/org/gnode/nix/Block.java]: Maybe it is nicer to handle removal in the normal setter:

```java
void setDefinition(String definition) {
    if (definition != null) {
        definition(definition);
    } else {
        definition(new None());
    }
}
```

File: src/main/java/org/gnode/nix/File.java [hunk 2]
```diff
@@ -89,6 +89,112 @@ public static File open(@StdString String name, int mode) {
...
+    @StdVector
+    Block blocks();
+
+    /**
+     * Get all blocks within this file.
+     *
+     * @return A list with Block entities.
+     * @see Block
+     */
+    public ArrayList<Block> getBlocks() {
```
Comment (stoewer) [inline src/main/java/org/gnode/nix/File.java]: Please just use `List<Block>`.

**Comment by sujithvm** [comment]
@stoewer I have made the required changes. Thank you

**Pull request merged**